planner/core/errors.go

```diff
@@ -19,74 +19,73 @@ import (
-	codeBadTable                        = mysql.ErrBadTable
-	codeKeyDoesNotExist                 = mysql.ErrKeyDoesNotExist
-	codeOperandColumns                  = mysql.ErrOperandColumns
-	codeInvalidWildCard                 = mysql.ErrParse
-	codeInvalidGroupFuncUse             = mysql.ErrInvalidGroupFuncUse
-	codeIllegalReference                = mysql.ErrIllegalReference
-	codeNoDB                            = mysql.ErrNoDB
-	codeUnknownExplainFormat            = mysql.ErrUnknownExplainFormat
-	codeWrongGroupField                 = mysql.ErrWrongGroupField
-	codeDupFieldName                    = mysql.ErrDupFieldName
-	codeNonUpdatableTable               = mysql.ErrNonUpdatableTable
-	codeInternal                        = mysql.ErrInternal
-	codeMixOfGroupFuncAndFields         = mysql.ErrMixOfGroupFuncAndFields
-	codeNonUniqTable                    = mysql.ErrNonuniqTable
-	codeWrongNumberOfColumnsInSelect    = mysql.ErrWrongNumberOfColumnsInSelect
-	codeWrongValueCountOnRow            = mysql.ErrWrongValueCountOnRow
-	codeTablenameNotAllowedHere         = mysql.ErrTablenameNotAllowedHere
-	codePrivilegeCheckFail              = mysql.ErrUnknown
-	codeWindowInvalidWindowFuncUse      = mysql.ErrWindowInvalidWindowFuncUse
-	codeWindowInvalidWindowFuncAliasUse = mysql.ErrWindowInvalidWindowFuncAliasUse
-	codeWindowNoSuchWindow              = mysql.ErrWindowNoSuchWindow
-	codeWindowCircularityInWindowGraph  = mysql.ErrWindowCircularityInWindowGraph
-	codeWindowNoChildPartitioning       = mysql.ErrWindowNoChildPartitioning
-	codeWindowNoInherentFrame           = mysql.ErrWindowNoInherentFrame
-	codeWindowNoRedefineOrderBy         = mysql.ErrWindowNoRedefineOrderBy
-	codeWindowDuplicateName             = mysql.ErrWindowDuplicateName
-	codeErrTooBigPrecision              = mysql.ErrTooBigPrecision
-	codePartitionClauseOnNonpartitioned = mysql.ErrPartitionClauseOnNonpartitioned
-	codeDBaccessDenied                  = mysql.ErrDBaccessDenied
-	codeTableaccessDenied               = mysql.ErrTableaccessDenied
-	codeSpecificAccessDenied            = mysql.ErrSpecificAccessDenied
-	codeViewNoExplain                   = mysql.ErrViewNoExplain
-	codeWindowFrameStartIllegal         = mysql.ErrWindowFrameStartIllegal
-	codeWindowFrameEndIllegal           = mysql.ErrWindowFrameEndIllegal
-	codeWindowFrameIllegal              = mysql.ErrWindowFrameIllegal
-	codeWindowRangeFrameOrderType       = mysql.ErrWindowRangeFrameOrderType
-	codeWindowRangeFrameTemporalType    = mysql.ErrWindowRangeFrameTemporalType
-	codeWindowRangeFrameNumericType     = mysql.ErrWindowRangeFrameNumericType
-	codeWindowRangeBoundNotConstant     = mysql.ErrWindowRangeBoundNotConstant
-	codeWindowRowsIntervalUse           = mysql.ErrWindowRowsIntervalUse
-	codeWindowFunctionIgnoresFrame      = mysql.ErrWindowFunctionIgnoresFrame
-	codeUnsupportedOnGeneratedColumn    = mysql.ErrUnsupportedOnGeneratedColumn
+	codeUnsupportedType                 terror.ErrCode = mysql.ErrUnsupportedType
```

Comment:
Could we avoid this part?

Reply:
Sorry I didn't get what you mean? Do you mean we want to totally get ride of `codeXXX` and just use `mysql.ErrXX`?

Reply:
yes